BUILD.plz

```diff
@@ -61,6 +61,171 @@ sh_cmd(
     cmd = "docker-compose stop",
 )
 
+sh_cmd(
+    name = "snapshot",
+    shell = "/usr/bin/env bash",
+    cmd = """
+set -o pipefail
+
+# Global regex variables
+NAT="0|[1-9][0-9]*"
+VERSION_REGEX="^[vV]?(\\\$NAT)\.(\\\$NAT)\.(\\\$NAT)$"
+PATCH_BRANCH_REGEX="^[vV]?(\\\$NAT)\.(\\\$NAT)\.x$"
+
+PROG=snapshot
+
+USAGE="\\
+Usage:
+  \\\$PROG (<tag>)
+  \\\$PROG --help
+
+Arguments:
+  <tag>      A tag for the final artifact of the snapshot (ie. Docker image).
+
+Options:
+  -r, --ref              The ref to create the tag for.
+  -n, --dry-run          Print the commands to be executed.
+  -h, --help             Print this help message.
+
+Tags can automatically be determined for the master branch (next minor)
+and patch branches matching the "*.*.x" pattern (next patch).
+
+For other branches and refs, provide a tag name.
+
+The command automatically detects the current ref,
+but you can also manually choose one using the --ref option.
+When the tag is automatically detected (see above),
+manually provided refs are ignored and are auto-detected."
+
+function info {
+  echo -e "\\\$1" >&2
+}
+
+function warn {
+  echo -e "\x1B[33m\\\$1\x1B[0m" >&2
+}
+
+function error {
+  echo -e "\x1B[31m\\\$1\x1B[0m" >&2
+  exit 1
+}
+
+# Default values of arguments
+REF=""
+DRY_RUN=false
+HELP=false
+ARGS=()
+
+# Loop through arguments and process them
+for arg in "\\\$@"
+do
+    case \\\$arg in
+        -r|--ref)
+        REF="\\\$2"
+        shift 2 # Remove argument name and value from processing
+        ;;
+        -n|--dry-run)
+        DRY_RUN=true
+        shift # Remove --dry-run from processing
+        ;;
+        -h|--help)
+        HELP=true
+        shift # Remove --help from processing
+        ;;
+        *)
+        ARGS+=("\\\$1")
+        shift # Remove generic argument from processing
+        ;;
+    esac
+done
+
+# Setting here so the last shift won't fail the script
+set -o errexit
```

Comment:
**Optional**: we **MAY** consider moving this closer to the `# Auto-detect tag (if possible)` conditional statement to make modifications to the `plz` command less error-prone thus avoiding the introduction of non-failing pipe commands between the setting and the mentioned conditional statement when those new commands are supposed to fail on pipe failure, but the author and reviewer miss the setting.

In addition I would consider turning pipefail off for the mentioned conditional block only and reenable pipefail right after that.